### PR TITLE
dep: update consul-template to v0.26.0

### PR DESCRIPTION
**Pull request by calvn**
Updates consul-template@v026.0

```
$ go get github.com/hashicorp/consul-template@v0.26.0
$ go mod vendor
$ go mod tidy
```

### CODE REVIEW

**Comment by calvn** [comment]
Closes https://github.com/hashicorp/vault/issues/9816

File: changelog/11830.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+agent: fix timestamp format in log messages from the templating engine
+```
```
Comment (tvoran) [inline changelog/11830.txt:3]: Might also be relevant to vault to add an entry for `fix command shell quoting issue [GH-1477, GH-1456, GH-1463]` from the 0.26.0 changlog?

**Review by tvoran — APPROVED** [review]

File: changelog/11830.txt (hunk 1 above)
Comment (calvn) [reply changelog/11830.txt:3]: Good point, I'll get that added!

**Comment by calvn** [comment]
Closing in favor of https://github.com/hashicorp/vault/pull/11838 due to removal of the `vendor/` dir

**Pull request closed without merging**